scipy/special/tests/test_basic.py

```diff
@@ -3610,6 +3611,24 @@ def test_riccati_yn(self):
         assert_array_almost_equal(C, special.riccati_yn(n, x), 8)
 
 
+class TestSoftplus:
+    @pytest.mark.parametrize("value, expected", [
+        (0, 0.6931471805599453),
+        ([-1, 0, 1], np.array([0.31326169, 0.69314718, 1.31326169])),
+        (100, 100.0),
+        (-5, 0.00671534848911849),
+        (-100, 0.0),
+        (10000, 10000.0)
+    ])
+    def test_softplus(self, value, expected):
+        
+        # Test cases for the softplus function.
+        # Note : ``value`` is selected based on the intervals provided in Eq.(10) of the the following paper:
+        # Mächler, M. (2012). log1mexp-note.pdf. Rmpfr: R MPFR - Multiple Precision Floating-Point Reliable.
+        # Retrieved from https://cran.r-project.org/web/packages/Rmpfr/vignettes/log1mexp-note.pdf
+        result = softplus(value)
+        assert np.allclose(result, expected)
+
```

Comment:
Test coverage: can you please add a basic test using a couple of the common keywords? E.g.:
```python
x = np.arange(5) - 2
out = np.ones(5)
special.softplus(x, where=x>0, out=out)
assert_allclose(out, np.array([1., 1., 1., 1.31326169, 2.12692801]))
```